File: functions/libsodium/variants/c/3_b64_urlsafe_char_to_byte_5.c
#include <stdint.h>

static unsigned int b64_urlsafe_char_to_byte(int c)
{
    if ((c >= 'A') && (c <= 'Z'))
        return (unsigned int)(c - 'A');
    if ((c >= 'a') && (c <= 'z'))
        return (unsigned int)(c - 'a') + 26;
    if ((c >= '0') && (c <= '9'))
        return (unsigned int)(c - '0') + 52;
    if (c == '-')
        return 62;
    if (c == '_')
        return 63;
    return 0;
}


int main(){
    b64_urlsafe_char_to_byte(0);
}

File: functions/libsodium/variants/go/3_b64_urlsafe_char_to_byte_5.go
package main

func b64_urlsafe_char_to_byte(c int32) uint {
    var result uint
    switch {
    case c >= 'A' && c <= 'Z':
        result = uint(c - 'A')
    case c >= 'a' && c <= 'z':
        result = uint(c - 'a' + 26)
    case c >= '0' && c <= '9':
        result = uint(c - '0' + 52)
    case c == '-':
        result = 62
    case c == '_':
        result = 63
    default:
        result = 0
    }
    return result
}


func main(){
    b64_urlsafe_char_to_byte(0)
}
